src/api/FactoryAPI.ts

```diff
@@ -138,6 +144,77 @@ export class FactoryAPI {
     return new BigNumber(10 ** (18 - minDecimal.toNumber()));
   }
 
+  public async calculateRequiredComponentUnits(
+    componentPrices: BigNumber[],
+    components: Address[],
+    componentAllocation: BigNumber[],
+    targetSetPrice: BigNumber,
+  ): Promise<BigNumber[]> {
+    this.assert.common.proportionsSumToOne(componentAllocation, coreAPIErrors.PROPORTIONS_DONT_ADD_UP_TO_1());
+    this.assert.common.isEqualLength(
+      componentPrices,
+      components,
+      coreAPIErrors.ARRAYS_EQUAL_LENGTHS('componentPrices', 'components')
+    );
+    this.assert.common.isEqualLength(
+      componentPrices,
+      componentAllocation,
+      coreAPIErrors.ARRAYS_EQUAL_LENGTHS('componentPrices', 'componentAllocation')
+    );
+
+    const targetComponentValues = this.calculateTargetComponentValues(componentAllocation, targetSetPrice);
+
+    // Calculate the target amount of tokens required by dividing the target component price
+    // with the price of a component and then multiply by the token's base unit amount
+    // (10 ** componentDecimal) to get it in base units.
+    const componentAmountRequiredPromises = targetComponentValues.map(async(targetComponentValue, i) => {
```

Comment:
space between async and the args. also i think we can use `_.map` instead since i think we use that a bit more  (i had just forgotten to use it in my original PR lol)